src/app/main.c

```diff
@@ -434,13 +456,9 @@ main (int    argc,
       g_dbus_error_strip_remote_error (local_error);
       g_printerr ("%serror: %s%s\n", prefix, suffix, local_error->message);
       g_error_free (local_error);
-
-      /* Print a warning if the exit status indicates success when we
-       * actually had an error, so it gets reported and fixed quickly. */
-      g_warn_if_fail (exit_status != EXIT_SUCCESS);
```

Comment:
Hmm, can we have still some sort of check here (or maybe in the `if` block at line 432)? It shouldn't be possible to have an error yet still exit 0, right?